query/src/main/java/org/infinispan/search/mapper/log/impl/Log.java

```diff
@@ -39,4 +42,19 @@ public interface Log extends BasicLogger {
    SearchException invalidEntitySuperType(String entityName,
          @FormatWith(ClassFormatter.class) Class<?> expectedSuperType,
          @FormatWith(ClassFormatter.class) Class<?> actualJavaType);
+
+   @LogMessage(level = WARN)
+   @Message(id = 14507, value = "Data is volatile, while indexes are stored on filesystem, " +
+         "in the meantime the index startup mode configuration is set to none: " +
+         "this setting could lead to some inconsistency between the indexes and the data " +
+         "in case of restarting the nodes.")
+   void volatileDataPersistentIndexesStartupModeNone();
+
+   @LogMessage(level = WARN)
+   @Message(id = 14508, value = "Data is persistent, while indexes are volatile, " +
+         "in the meantime the index startup mode configuration is set to none or purge: " +
+         "this setting could lead to some inconsistency between the indexes and the data " +
+         "in case of restarting the nodes.")
+   void persistentDataVolatileIndexesStartupModeNoneOrPurge();
```

Comment:
you could use a single method:
```java
logIndexStartupModeMismatch(String data, String index, String startupMode)
```
with message
`Data is %s, while indexes are %s, in the meantime the index startup mode configuration is set to %s...`